frontend/src/components/App/Home/index.tsx

```diff
@@ -0,0 +1,213 @@
+      >
+        <Icon icon="mdi:more-vert" />
+      </IconButton>
+      <Menu
+        id={menuId}
+        anchorEl={anchorEl}
+        keepMounted
+        open={Boolean(anchorEl)}
+        onClose={() => {
+          handleMenuClose();
+        }}
+      >
+        <MenuItem
+          onClick={() => {
+            history.push(createRouteURL('cluster', { cluster: cluster.name }));
+            handleMenuClose();
+          }}
+        >
+          <ListItemText>{t('settings|View')}</ListItemText>
+        </MenuItem>
+        <MenuItem
+          onClick={() => {
+            history.push(createRouteURL('settingsCluster', { cluster: cluster.name }));
+            handleMenuClose();
+          }}
+        >
+          <ListItemText>{t('settings|Settings')}</ListItemText>
+        </MenuItem>
+        {helpers.isElectron() && cluster.meta_data?.source === 'dynamic_cluster' && (
+          <MenuItem
+            onClick={() => {
+              removeCluster(cluster);
+              handleMenuClose();
+            }}
+          >
+            <ListItemText>{t('frequent|Delete')}</ListItemText>
+          </MenuItem>
+        )}
+      </Menu>
+    </>
+  );
+}
+
+function ClusterStatus({ error }: { error?: ApiError | null }) {
+  const { t } = useTranslation(['cluster', 'frequent']);
+  const theme = useTheme();
+
+  const stateUnknown = error === undefined;
+  const hasReachError = error && error.status !== 401 && error.status !== 403;
+
+  return (
+    <Box width="fit-content">
+      <Box display="flex" alignItems="center" justifyContent="center">
+        {hasReachError ? (
+          <Icon icon="mdi:cloud-off" width={16} color={theme.palette.home.status.error} />
+        ) : stateUnknown ? (
+          <Icon icon="mdi:cloud-question" width={16} color={theme.palette.home.status.unknown} />
+        ) : (
+          <Icon
+            icon="mdi:cloud-check-variant"
+            width={16}
+            color={theme.palette.home.status.success}
+          />
+        )}
+        <Typography
+          variant="body2"
+          style={{
+            marginLeft: theme.spacing(1),
+            color: hasReachError
+              ? theme.palette.home.status.error
+              : !stateUnknown
+              ? theme.palette.home.status.success
+              : undefined,
+          }}
+        >
+          {hasReachError ? error.message : stateUnknown ? '⋯' : t('frequent|Active')}
+        </Typography>
+      </Box>
+    </Box>
+  );
+}
+
+export default function Home() {
+  const history = useHistory();
+  const clusters = useClustersConf() || {};
+
+  if (!helpers.isElectron() && Object.keys(clusters).length === 1) {
+    history.push(createRouteURL('cluster', { cluster: Object.keys(clusters)[0] }));
+    return null;
+  }
+
+  return <HomeComponent clusters={clusters} />;
+}
+
+interface HomeComponentProps {
+  clusters: { [name: string]: Cluster };
+}
+
+function HomeComponent(props: HomeComponentProps) {
```

Comment:
This could also be in Home/Home.tsx, perhaps named PureHome rather than HomeComponent.

Reply:
This is not a PureHome style component. I don't think we should do those actually.
The reason for this division is just so we can call a hook sort of conditionally (if we are not in electron or the clusters are less than 2, we don't want to check for their versions).